src/system.js

```diff
@@ -326,3 +472,41 @@ module.exports = AFRAME.registerSystem('physics', {
     return this.driver.getMaterial(name);
   }
 });
+
+class StatTracker {
+  constructor(cycles = 100, dps = 2) {
+    this.cycles = cycles
+    this.dps = dps
+    this.reset()
+  }
+
+  reset() {
+    this.max = 0
+    this.min = Infinity
+    this.cumulative = 0
+    this.counter = 0
+  }
+
+  record(value) {
+    this.cumulative += value
+    if (value > this.max) {
+      this.max = value
+    }
+    if (value < this.min) {
+      this.min = value
+    }
+    this.counter++
+  }
+
+  get avg() {
+    // Average will only be valid if we have recorded the correct number of cycles.
+    console.assert(this.counter === this.cycles)
+    return this.cumulative / this.cycles
+  }
+
+  get report() {
+    return { avg: this.avg.toFixed(this.dps),
```

Comment:
new object generation, could be recycled

Reply:
This code is now in aframe-stats-panel.  And I fixed this issue.